yarn_api_client/hadoop_conf.py

```diff
@@ -47,33 +43,28 @@ def _get_resource_manager(hadoop_conf_path, rm_id=None):
     return rm_webapp_address or None
 
 
-def check_is_active_rm(url, timeout=30):
-    uri = Uri(url)
-    if uri.is_https:
-        conn = HTTPSConnection(host=uri.hostname, port=uri.port, timeout=timeout)
-    else:
-        conn = HTTPConnection(host=uri.hostname, port=uri.port, timeout=timeout)
+def check_is_active_rm(url, timeout=30, auth=None, verify=True):
     try:
-        conn.request('GET', '/cluster')
+        response = requests.get(url + "/cluster", timeout=timeout, auth=auth, verify=verify)
     except:
         return False
-    response = conn.getresponse()
-    if response.status != OK:
+
+    if response.status_code == 401:
+        raise Exception("401 Unauthorized")
```

Comment:
Per your previous question, could it be the case that in a multi-RM config, one RM may produce a `401`, while the other doesn't?  If so, then I suppose that would argue for returning `False` here.  On the other hand, and in most cases, both RMs would produce `401` and raising 'No active RMs found' would lose the context of authorization if we returned `False` here.  

I think in either case, it might be good to log an error with the url here.  If we returned `False`, then there would likely be two log messages indicating an unauthorized attempt.

I guess after typing this out, I'm leaning toward returning `False` with a log message.  Perhaps it might help to update the ultimate exception text to something like `No active resource managers are accessible at this time.`, then the admin will see the authorization errors in the log.

I wonder if we should log in both cases and just change to `!= 200`, but also add the `response.message` to the logged statement?  I'm fine either way, but starting to wonder if they get the exception, but don't have any additional information (for the non-active case) then that doesn't seem helpful - so logging on any error might be better.

What do you think?